stl/inc/format

```diff
@@ -385,15 +386,106 @@ _NODISCARD constexpr const _CharT* _Parse_arg_id(
     throw format_error("Invalid format string.");
 }
 
+inline int _Code_units_in_next_character(const char* _First, const char* _Last, const _Cvtvec& _Cvt) {
+    // Returns a count of the number of code units (characters) that compose the first multibyte
+    // character in [_First, _Last), or -1 if [_First, _Last) doesn't contain an entire encoded
+    // character or *_First is not a valid lead byte.
+    _STL_INTERNAL_CHECK(_First < _Last);
+
+    switch (_Cvt._Mbcurmax) {
+    default:
+        _ASSERTE(!"Bad number of encoding units for this code page");
+        [[fallthrough]];
+    case 1:
+        return 1; // all characters have only one code unit
+
+    case 2:
+        {
+            wchar_t _Wide;
+            mbstate_t _St{};
+            const auto _Len   = static_cast<size_t>(_Last - _First);
+            const int _Result = _Mbrtowc(&_Wide, _First, _Len, &_St, &_Cvt);
+            if (_Result > 0) {
+                return _Result;
+            } else if (_Result < 0) { // invalid or incomplete encoded character
+                return -1;
+            } else { // next code unit is '\0'
+                return 1;
+            }
+        }
+
+    case 4: // Assume UTF-8 (as does _Mbrtowc)
+        {
+            const auto _Ch = static_cast<unsigned char>(*_First);
+            if (_Ch < 0b1000'0000) {
+                return 1;
+            }
+
+            const auto _Len = static_cast<size_t>(_Last - _First);
+
+            if (_Ch < 0b1110'0000) {
+                if (_Ch >= 0b1100'0000 && _Len >= 2) {
+                    return 2;
+                }
+
+                // non-lead byte or partial code unit
+                return -1;

```

Comment:
I was trying to avoid triggering Charlie's dislike of the conditional operator, but I agree that there's a ton of wasted space here. I'll see if I can tighten things up with some conditonals.